core-web/libs/template-builder/src/lib/components/template-builder/components/add-style-classes-dialog/add-style-classes-dialog.component.ts

```diff
@@ -0,0 +1,111 @@
+import { Subject } from 'rxjs';
+
+import { AfterViewInit, ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+
+import { AutoCompleteModule } from 'primeng/autocomplete';
+import { ButtonModule } from 'primeng/button';
+import { DynamicDialogConfig, DynamicDialogRef } from 'primeng/dynamicdialog';
+
+import { StyleClassModel } from '../../models/models';
+
+@Component({
+    selector: 'dotcms-add-style-classes-dialog',
+    standalone: true,
+    imports: [AutoCompleteModule, FormsModule, ButtonModule],
+    templateUrl: './add-style-classes-dialog.component.html',
+    styleUrls: ['./add-style-classes-dialog.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class AddStyleClassesDialogComponent implements OnInit, AfterViewInit {
+    classes: StyleClassModel[];
+
+    selectedClasses: StyleClassModel[] = [];
+
+    filteredClasses: StyleClassModel[];
+
+    private autoCompleteInput: HTMLInputElement;
+
+    constructor(
+        private ref: DynamicDialogRef,
+        public dynamicDialogConfig: DynamicDialogConfig<{
+            classes: Subject<StyleClassModel[]>;
+            selectedClasses: string[];
+        }>
+    ) {}
+
+    ngOnInit() {
+        const { classes: classes$, selectedClasses } = this.dynamicDialogConfig.data;
+
+        classes$.subscribe((classes) => {
+            this.classes = classes;
+        });
+
+        this.selectedClasses = selectedClasses.map((klass) => ({
+            klass
+        }));
+    }
+
+    ngAfterViewInit() {
+        this.autoCompleteInput = document.getElementById('my-autocomplete') as HTMLInputElement;
+    }
+    /**
+     * @description Filters the classes based on the query
+     *
+     * @param {{ query: string }} { query }
+     * @return {*}
+     * @memberof AddStyleClassesDialogComponent
+     */
+    filterClasses({ query }: { query: string }): void {
+        const filtered: StyleClassModel[] = [];
+
+        if ((query.includes(',') || query.includes(' ')) && query.trim().length > 0) {
```

Comment:
the query.trim().length works as a short circuit, so it should be the first sentence on the conditional